quarry/web/worker.py

```diff
@@ -22,17 +22,18 @@
 celery_log = get_task_logger(__name__)
 
 celery = Celery("quarry.web.worker")
-celery.conf.update(
-    yaml.safe_load(open(os.path.join(__dir__, "../default_config.yaml")))
-)
 try:
     celery.conf.update(
-        yaml.safe_load(open(os.path.join(__dir__, "../config.yaml")))
+        yaml.safe_load(open(os.path.join(__dir__, "/config/config.yaml")))
```

Comment:
This path seems to exist only with the k8s setup?

Also, the code seems to be duplicated across 3 files, we should probably put this in a utils file.

Reply:
This is true. I believe the original assumption was that there would be no VM, thus no need to manage where its config file would be. Though this may not be true in the immediate term. So we could copy the config.yaml on the VMs to "/config/" or add another try/except, though that seems kind of icky, as that would result in a:
```
try:
    celery.conf.update(
        yaml.safe_load(open(os.path.join(__dir__, "/config/config.yaml")))
        skip="true"
    )
except IOError:
    pass

if skip:
  try:
      # for pytest
      celery.conf.update(
          yaml.safe_load(open(os.path.join(__dir__, "../default_config.yaml")))
      )
  except IOError:
      celery.conf.update(
          yaml.safe_load(open(os.path.join(__dir__, "../config.yaml")))
      )
      pass
```
Which seems quite verbose. Though I suspect there is a cleaner way of doing that which eludes me. What route do you recommend?

And I agree it should be in a shared file rather than spread across three files. Though I feel that change should be a separate ticket. I opened https://phabricator.wikimedia.org/T349135 to track

Reply:
Perhaps something like the current commit?

Reply:
I don't think we need conditions to distinguish VM/k8s/dev/test setups. Couldn't we mount both the `config.yaml` and `default_config.yaml` files (as opposed to mounting the config directory) and retain the existing logic where we read from default_config and apply overrides from config.yaml?

Reply:
Oh neat, I didn't realize that a configmap could be mounted as a single file. Thank you for pointing that out.